library/src/scala/tasty/Reflection.scala

```diff
@@ -6,329 +6,459 @@ import scala.quoted.QuoteContext
+ *           |               +- Refined
+ *           |               +- Applied
+ *           |               +- Annotated
+ *           |               +- MatchTypeTree
+ *           |               +- ByName
+ *           |               +- LambdaTypeTree
+ *           |               +- TypeBind
+ *           |               +- TypeBlock
+ *           |
+ *           +- TypeBoundsTree
+ *           +- WildcardTypeTree
+ *           |
+ *           +- CaseDef
+ *           |
+ *           +- TypeCaseDef
+ *           +- Bind
+ *           +- Unapply
+ *           +- Alternatives
+ *
+ *
+ *  +- Type -+- ConstantType
+ *           +- TermRef
+ *           +- TypeRef
+ *           +- SuperType
+ *           +- Refinement
+ *           +- AppliedType
+ *           +- AnnotatedType
+ *           +- AndType
+ *           +- OrType
+ *           +- MatchType
+ *           +- ByNameType
+ *           +- ParamRef
+ *           +- ThisType
+ *           +- RecursiveThis
+ *           +- RecursiveType
+ *           +- LambdaType -+- MethodType
+ *           |              +- PolyType
+ *           |              +- TypeLambda
+ *           +- TypeBounds
+ *           +- NoPrefix
+ *
+ *  +- ImportSelector -+- SimpleSelector
+ *                     +- RenameSelector
+ *                     +- OmitSelector
+ *
+ *  +- Id
+ *
+ *  +- Signature
+ *
+ *  +- Position
+ *
+ *  +- Comment
+ *
+ *  +- Constant
+ *
+ *  +- Symbol
+ *
+ *  +- Flags
+ *
+ *  ```
  */
-trait Reflection extends reflect.Types { reflectSelf: CompilerInterface =>
+trait Reflection { reflection =>
 
   //////////////
   // CONTEXTS //
   //////////////
 
+  /** Compilation context */
+  type Context <: AnyRef
+
   /** Context of the macro expansion */
   def rootContext: Context // TODO: Should this be moved to QuoteContext?
   given Context = rootContext // TODO: Should be an implicit converion from QuoteContext to Context
 
-
   ///////////////
-  //   Source  //
+  //   TREES   //
   ///////////////
 
-  object Source:
-
-    /** Returns the source file being compiled. The path is relative to the current working directory. */
-    def path: java.nio.file.Path = reflectSelf.Source_path
-
-    /** Returns true if we've tried to reflect on a Java class. */
-    def isJavaCompilationUnit: Boolean = reflectSelf.Source_isJavaCompilationUnit
 
-    /** Returns true if we've tried to reflect on a Scala2 (non-Tasty) class. */
-    def isScala2CompilationUnit: Boolean = reflectSelf.Source_isScala2CompilationUnit
+  /** Tree representing code written in the source */
+  type Tree <: AnyRef
 
-    /** Returns true if we've tried to reflect on a class that's already loaded (e.g. Option). */
-    def isAlreadyLoadedCompilationUnit: Boolean = reflectSelf.Source_isAlreadyLoadedCompilationUnit
+  val Tree: TreeModule
 
-    /** Class name of the current CompilationUnit */
-    def compilationUnitClassname: String = reflectSelf.Source_compilationUnitClassname
+  trait TreeModule { this: Tree.type => }
```

Comment:
Possibly. If we do, it is better to have it already defined to avoid breaking user code if we add it later. This happened in the past with the `Position` module.